src/test/scala/chiselTests/LTLSpec.scala

```diff
@@ -407,4 +407,21 @@ class LTLSpec extends AnyFlatSpec with Matchers with ChiselRunners {
     assert(assertBlockLoc < delayIntrinsicLoc)
     assert(assumeblockLoc < implicationIntrinsicLoc)
   }
+
+  it should "not produce name collisions with clock" in {
```

Comment:
I added this test to reproduce the bug. It seems like a silly test now that there is no possibility for a name collision. However, it also felt wrong to delete a test, even if it is trivially passing.

Should we keep it?

Reply:
We should keep it, but possibly extend it to check the names that come out

Reply:
Sounds good, will add the extended test shortly